rm-datasource/src/main/java/io/seata/rm/datasource/exec/LockRetryController.java

```diff
@@ -84,7 +87,50 @@ private int getLockRetryTimes() {
             }
         }
         // if there is no customized config, use global config instead
-        Configuration configuration = ConfigurationFactory.getInstance();
-        return configuration.getInt(ConfigurationKeys.CLIENT_LOCK_RETRY_TIMES, DEFAULT_CLIENT_LOCK_RETRY_TIMES);
+        return globalConfig.getGlobalLockRetryTimes();
+    }
+
+    static class GlobalConfig implements ConfigurationChangeListener {
+
+        private int globalLockRetryInternal;
+
+        private int globalLockRetryTimes;
+
+        private final int defaultRetryInternal = DefaultValues.DEFAULT_CLIENT_LOCK_RETRY_INTERVAL;
+        private final int defaultRetryTimes = DefaultValues.DEFAULT_CLIENT_LOCK_RETRY_TIMES;
+
+        public GlobalConfig() {
+            Configuration configuration = ConfigurationFactory.getInstance();
+            globalLockRetryInternal = configuration.getInt(ConfigurationKeys.CLIENT_LOCK_RETRY_INTERVAL, defaultRetryInternal);
+            globalLockRetryTimes = configuration.getInt(ConfigurationKeys.CLIENT_LOCK_RETRY_TIMES, defaultRetryTimes);
+        }
+
+        @Override
+        public void onChangeEvent(ConfigurationChangeEvent event) {
+            String dataId = event.getDataId();
+            String newValue = event.getNewValue();
+            if (ConfigurationKeys.CLIENT_LOCK_RETRY_INTERVAL.equals(dataId)) {
+                globalLockRetryInternal = parseInt(newValue, defaultRetryInternal);
+            }
+            if (ConfigurationKeys.CLIENT_LOCK_RETRY_TIMES.equals(dataId)) {
+                globalLockRetryTimes = parseInt(newValue, defaultRetryTimes);
+            }
+        }
+
+        private int parseInt(String value, int fallback) {
```

Comment:
use the static method `NumberUtils.toInt`

Reply:
ok, i see it.